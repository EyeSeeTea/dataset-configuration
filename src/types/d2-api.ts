import { D2Api } from "@eyeseetea/d2-api/2.36";
import { getMockApiFromClass } from "@eyeseetea/d2-api";

export { CancelableResponse } from "@eyeseetea/d2-api";
export { D2Api } from "@eyeseetea/d2-api/2.36";
export type { MetadataPick } from "@eyeseetea/d2-api/2.36";
export const getMockApi = getMockApiFromClass(D2Api);

interface LocalInstance {
    type: "local";
    url: string;
}

interface ExternalInstance {
    type: "external";
    url: string;
    username: string;
    password: string;
}

export type DhisInstance = LocalInstance | ExternalInstance;

export function getD2APiFromInstance(instance: DhisInstance) {
    return new D2Api({
        baseUrl: instance.url,
        auth:
            instance.type === "external"
                ? { username: instance.username, password: instance.password }
                : undefined,
        backend: "fetch",
    });
}
