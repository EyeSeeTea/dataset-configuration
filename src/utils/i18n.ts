import { getModuleForNamespace } from "./i18n-typed";

export default getModuleForNamespace("d2-dataset-configuration");
