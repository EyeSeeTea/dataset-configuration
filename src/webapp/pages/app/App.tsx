import styled from "styled-components";
import { HeaderBar } from "@dhis2/ui";
import { LoadingProvider, SnackbarProvider } from "@eyeseetea/d2-ui-components";
import { MuiThemeProvider } from "@material-ui/core/styles";
//@ts-ignore
import OldMuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import React, { useEffect, useState } from "react";
import { CompositionRoot } from "$/CompositionRoot";
import { AppContext, AppContextState } from "$/webapp/contexts/app-context";
import { Router } from "$/webapp/pages/Router";
import "./App.css";
import muiThemeLegacy from "./themes/dhis2-legacy.theme";
import { muiTheme } from "./themes/dhis2.theme";
import { D2Api } from "$/types/d2-api";

export interface AppProps {
    compositionRoot: CompositionRoot;
    api: D2Api;
}

function App(props: AppProps) {
    const { api, compositionRoot } = props;
    const [loading, setLoading] = useState(true);
    const [appContext, setAppContext] = useState<AppContextState | null>(null);

    useEffect(() => {
        async function setup() {
            const currentUser = await compositionRoot.users.getCurrent.execute().toPromise();
            if (!currentUser) throw new Error("User not logged in");

            setAppContext({ api, currentUser, compositionRoot });
            setLoading(false);
        }
        setup();
    }, [compositionRoot, api]);

    if (loading) return null;

    return (
        <MuiThemeProvider theme={muiTheme}>
            <OldMuiThemeProvider muiTheme={muiThemeLegacy}>
                {/* @ts-ignore */}
                <LoadingProvider>
                    <SnackbarProvider>
                        <StyledHeaderBar appName="Project Configuration app" />

                        <div id="app" className="content">
                            <AppContext.Provider value={appContext}>
                                <Router />
                            </AppContext.Provider>
                        </div>
                    </SnackbarProvider>
                </LoadingProvider>
            </OldMuiThemeProvider>
        </MuiThemeProvider>
    );
}

const StyledHeaderBar = styled(HeaderBar)`
    div:first-of-type {
        box-sizing: border-box;
    }
`;

export default React.memo(App);
