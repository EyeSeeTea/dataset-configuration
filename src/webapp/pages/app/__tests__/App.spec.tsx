import { render } from "@testing-library/react";

import App from "$/webapp/pages/app/App";
import { getTestContext } from "$/utils/tests";
import { Provider } from "@dhis2/app-runtime";
import { getD2APiFromInstance } from "$/types/d2-api";

describe("App", () => {
    it("navigates to page", async () => {
        const view = getView();

        expect(await view.findByText("No results found")).toBeInTheDocument();
    });
});

function getView() {
    const { compositionRoot } = getTestContext();
    const baseUrl = "http://localhost:8080";
    const api = getD2APiFromInstance({ type: "local", url: baseUrl });
    return render(
        <Provider config={{ baseUrl: "http://localhost:8080", apiVersion: 30 }}>
            <App api={api} compositionRoot={compositionRoot} />
        </Provider>
    );
}
