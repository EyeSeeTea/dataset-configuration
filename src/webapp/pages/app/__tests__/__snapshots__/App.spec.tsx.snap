// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > navigates to page 1`] = `
<DocumentFragment>
  <header
    class="jsx-2837859311 sc-dhKdPU klBWsW"
  />
  <div
    class="content"
    id="app"
  >
    <div>
      <button
        aria-label="Back"
        class="MuiButtonBase-root MuiIconButton-root sc-gEvDqW sc-eqUzNf hwoczq haqPoX MuiIconButton-colorSecondary"
        data-test="page-header-back"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root MuiIcon-colorPrimary"
          >
            arrow_back
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h5
        class="MuiTypography-root sc-aYaIB hZQTlr MuiTypography-h5 MuiTypography-gutterBottom"
        data-test="page-header-title"
      >
        Detail page
      </h5>
    </div>
    <h2
      class="sc-fqkwJk lcQFEV"
    >
      Hello John
    </h2>
  </div>
</DocumentFragment>
`;
